website_sale_product_subscribe_notify/models/mail_mail.py

```diff
@@ -0,0 +1,179 @@
+
+                # headers
+                headers = {}
+                ICP = self.env['ir.config_parameter'].sudo()
+                bounce_alias = ICP.get_param("mail.bounce.alias")
+                catchall_domain = ICP.get_param("mail.catchall.domain")
+                if bounce_alias and catchall_domain:
+                    if mail.model and mail.res_id:
+                        headers['Return-Path'] = '%s+%d-%s-%d@%s' % (
+                            bounce_alias, mail.id, mail.model, mail.res_id, catchall_domain)
+                    else:
+                        headers['Return-Path'] = '%s+%d@%s' % (
+                            bounce_alias, mail.id, catchall_domain)
+                if mail.headers:
+                    try:
+                        headers.update(safe_eval(mail.headers))
+                    except Exception:
+                        pass
+
+                # Writing on the mail object may fail (e.g. lock on user) which
+                # would trigger a rollback *after* actually sending the email.
+                # To avoid sending twice the same email, provoke the failure earlier
+                mail.write({
+                    'state': 'exception',
+                    'failure_reason': _('Error without exception. Probably due do sending an email without computed recipients.'),
+                })
+                mail_sent = False
+
+                # Update notification in a transient exception state to avoid concurrent
+                # update in case an email bounces while sending all emails related to current
+                # mail record.
+                notifs = self.env['mail.notification'].search([
+                    ('is_email', '=', True),
+                    ('mail_message_id', 'in', mail.mapped('mail_message_id').ids),
+                    ('res_partner_id', 'in', mail.mapped('recipient_ids').ids),
+                    ('email_status', 'not in', ('sent', 'canceled'))
+                ])
+                if notifs:
+                    notifs.sudo().write({
+                        'email_status': 'exception',
+                    })
+
+                # build an RFC2822 email.message.Message object and send it without queuing
+                res = None
+                for email in email_list:
+                    msg = IrMailServer.build_email(
+                        email_from=mail.email_from,
+                        email_to=email.get('email_to'),
+                        subject=mail.subject,
+                        body=email.get('body'),
+                        body_alternative=email.get('body_alternative'),
+                        email_cc=tools.email_split(mail.email_cc),
+                        # <<< QTL EDIT
+                        # Strange that build_email() supports BCC but Odoo
+                        # does not add bcc in this method.
+                        email_bcc=tools.email_split(mail.email_bcc),
+                        # >>> QTL EDIT
```

Comment:
Would it be better to split this change into another module and let website_sale_product_subscribe_notify depend on that to make the solution more generic?

Reply:
That's true, it should be easier to understand the logic as well.
Let me follow up on this.